src/app/features/languageSlice.ts

```diff
@@ -0,0 +1,30 @@
+import { PayloadAction, createSlice } from '@reduxjs/toolkit';
+import { ILanguage } from '../../config';
+import i18n from '../../i18n/i18n';
+
+interface LanguageState {
+    currentLanguage: ILanguage;
+}
+
+const initialState: LanguageState = {
+    currentLanguage: i18n.language as ILanguage.English,
+};
+
+const languageSlice = createSlice({
+    name: 'i18n',
+    initialState,
+    reducers: {
+        toggleLanguage: (state, action: PayloadAction<ILanguage>) => {
+            state.currentLanguage = action.payload;
+            i18n.changeLanguage(action.payload)
+                .then(() => {})
```

Comment:
Is the .then() portion left empty for later use? Could add a comment for clarity.

Reply:
Resolved in b11ffac